payments-core/src/main/java/com/stripe/android/payments/core/authentication/DefaultIntentAuthenticatorRegistry.kt

```diff
@@ -0,0 +1,116 @@
+package com.stripe.android.payments.core.authentication
+
+import androidx.activity.result.ActivityResultLauncher
+import com.stripe.android.Logger
+import com.stripe.android.PaymentAuthConfig
+import com.stripe.android.PaymentBrowserAuthStarter
+import com.stripe.android.PaymentRelayStarter
+import com.stripe.android.StripePaymentController
+import com.stripe.android.model.StripeIntent
+import com.stripe.android.networking.AnalyticsRequestExecutor
+import com.stripe.android.networking.AnalyticsRequestFactory
+import com.stripe.android.networking.StripeRepository
+import com.stripe.android.payments.PaymentFlowResult
+import com.stripe.android.payments.core.injection.AuthenticationModule
+import com.stripe.android.payments.core.injection.DaggerAuthenticationComponent
+import com.stripe.android.stripe3ds2.service.StripeThreeDs2Service
+import com.stripe.android.stripe3ds2.transaction.MessageVersionRegistry
+import com.stripe.android.view.AuthActivityStarter
+import javax.inject.Inject
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * Default registry to provide look ups for [IntentAuthenticator].
+ * Should be only accessed through [DefaultIntentAuthenticatorRegistry.getInstance].
+ */
+internal class DefaultIntentAuthenticatorRegistry @Inject internal constructor() :
+    IntentAuthenticatorRegistry {
+
+    // default authenticators always packaged with the SDK
+    @Inject
+    lateinit var webIntentAuthenticator: WebIntentAuthenticator
+
+    @Inject
+    lateinit var noOpIntentAuthenticator: NoOpIntentAuthenticator
+
+    // authenticators requiring a 3p SDK
+    // TODO(ccen) move them to a dedicated module
+    @Inject
+    lateinit var stripe3DS2Authenticator: Stripe3DS2Authenticator
+
+    override fun lookUp(stripeIntent: StripeIntent): IntentAuthenticator {
+        if (!stripeIntent.requiresAction()) {
+            return noOpIntentAuthenticator
+        }
+
+        when (val nextActionData = stripeIntent.nextActionData) {
+            is StripeIntent.NextActionData.SdkData.Use3DS2 -> {
+                return stripe3DS2Authenticator
+            }
+            is StripeIntent.NextActionData.SdkData.Use3DS1 -> {
+                // can only triggered when `use_stripe_sdk=true`
+                return webIntentAuthenticator
+            }
+            is StripeIntent.NextActionData.RedirectToUrl -> {
+                // can only triggered when `use_stripe_sdk=false`
+                return webIntentAuthenticator
+            }
+            is StripeIntent.NextActionData.AlipayRedirect -> {
```

Comment:
For Alipay we support two types of authentications: web and SDK. 

This is the web path, triggered by `Stripe#confirmPayment`. It will be handled by the generic `WebIntentAuthenticator`  that's included in the main SDK and not be split further.

For the SDK path, currently triggered by `Stripe#confirmAlipayPayment`, the client will need to create an `AlipayAuthenticator`, calling back into Alipay SDK and pass to the API. That path is handled in [DefaultAlipayRepository#authenticate](https://github.com/stripe/stripe-android/blob/0e7f4dd7c881e6c058165df7bb00e59eb8570de5/payments-core/src/main/java/com/stripe/android/networking/DefaultAlipayRepository.kt#L25) and is not affected in the PR.

Currently the client use `Stripe#confirmPayment` for web auth and `Stripe#confirmAlipayPayment` for sdk auth, with [milestone 3.1](https://paper.dropbox.com/doc/diagram-Modularizing-Android-Mobile-SDK--BL3dd7hkbuzekCUeNFZ_NAWPAg-M9mrrTib7RFBVN7VimeGx), we'll be able to deprecate `Stripe#confirmAlipayPayment` and merge both in `Stripe#confirmPayment`  with a dedicated alipay module